tensorflow/contrib/seq2seq/python/ops/beam_search_decoder.py

```diff
@@ -297,11 +299,15 @@ def initialize(self, name=None):
     """
     finished, start_inputs = self._finished, self._start_inputs
 
+    log_probs = array_ops.one_hot(  # shape(batch_sz, beam_sz)
+        array_ops.zeros([self._batch_size], dtype=dtypes.int32),
+        depth=self._beam_width, on_value=0.0,
+        off_value=-np.Inf, dtype=dtypes.float32)
```

Comment:
make sure that this dtype matches that of the input.  it could probably also be one of float64 or float16.